remote-api/src/main/java/org/alfresco/rest/api/model/rules/RuleBuilder.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * #%L
+ * Alfresco Remote API
+ * %%
+ * Copyright (C) 2005 - 2022 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.rest.api.model.rules;
+
+/** Builder class for {@link Rule}. */
+public class RuleBuilder

```

Comment:
The builder could be also placed as a static inner class in the `Rule`. Then it could be initiated within factory method builder(), eg. `Rule.builder().id().name().create()`. In fact this is a matter of preferences, so I'm also ok with this approach.

Reply:
I spent a while trying to get IntelliJ to do this for me, but in the end gave up and generated the builder as a new class.  I'll move it to a static inner class, even though I hate inner classes for almost every purpose then I think this is neater (it's basically a constructor).